core/src/models/ExportProcessor.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  Model,
+  Table,
+  Column,
+  CreatedAt,
+  UpdatedAt,
+  AllowNull,
+  Is,
+  BelongsTo,
+  BeforeCreate,
+  BeforeSave,
+  ForeignKey,
+  DataType,
+  Default,
+  HasMany,
+} from "sequelize-typescript";
+import { Op, QueryTypes } from "sequelize";
+import { api, config } from "actionhero";
+import * as uuid from "uuid";
+import { Destination } from "./Destination";
+import { APIData } from "../modules/apiData";
+import { StateMachine } from "../modules/stateMachine";
+import { Export } from "./Export";
+import { ExportProcessorOps } from "../modules/ops/exportProcessor";
+
+const ERROR_LEVELS = ["error", "info"] as const;
+export type ExportProcessorErrorLevel = typeof ERROR_LEVELS[number];
+
+export const ExportProcessorStates = [
+  "pending", // waiting to be processed
+  "failed", // something went wrong and we won't try again
+  "complete", // OK!
+] as const;
+
+const STATE_TRANSITIONS = [
+  { from: "pending", to: "failed", checks: [] },
+  {
+    from: "pending",
+    to: "complete",
+    checks: [
+      (instance: ExportProcessor) => instance.ensureAllExportsProcessed(),
+    ],
+  },
+];
+
+@Table({ tableName: "exportProcessors", paranoid: false })
+export class ExportProcessor extends Model {
```

Comment:
Why not extend LoggedModel?

Reply:
The timestamps and id generation come from there, as well as logging

Reply:
Ah! I started off from using the `Export` model as a base and it apparently does not extend LoggedModel. Changed!